proxy.h

```diff
@@ -590,44 +590,81 @@ proxy<F> make_proxy(T&& value) {
 
 // The following types and macros aim to simplify definition of dispatch and
 // facade types prior to C++26
-namespace helper {
+namespace details {
+
+template <class Args, class O>
+struct one_overload_matching_traits : inapplicable_traits {};
+template <class Args0, class R, class... Args1>
+    requires(std::is_same_v<Args0, std::tuple<Args1&&...>>)
+struct one_overload_matching_traits<Args0, R(Args1...)> : applicable_traits {};
+template <class Args, class Os>
+struct overloads_matching_traits : inapplicable_traits {};
+template <class... Args, class... Os> requires(
+    one_overload_matching_traits<std::tuple<Args...>, Os>::applicable || ...)
```

Comment:
`&&`?

Reply:
Never mind. It should be `||`.

Reply:
It should be `||`. The semantics of `overloads_matching_traits` is that, for a given `Args` pack, any overload in `Os` matches the argument types.

Reply:
Is it an O(n) matching? or O(n2), or even worse?

Reply:
O(n), at compile time